### ISSUE TITLE
Implement DDN reference-pattern correlation in BOM Summary

### ISSUE THREAD

**Issue by jwsiejk** [jwsiejk/accountlist#368]
## Goal
Implement a DDN reference-pattern correlation layer in the AI Workload Discovery Mapper so the **BOM Summary** explains how each selected workload maps to DDN-informed architecture/reference patterns.

## Important intent
The BOM Summary should **not** generate a real BOM. It should explain, in plain English:

```text
Workload signal → Architecture pressure → Closest DDN reference pattern → Relevant building blocks → Validation before BOM
```

The purpose is to help James explain why a workload points toward certain DDN-style building blocks, using DDN public reference architecture/platform themes as the guide.

## Source-backed framing
Use DDN public documentation themes as the basis:

- DDN Financial Services page: trading, risk, fraud, compliance, real-time analytics, fraud detection, high-throughput/low-latency AI data strategies.
- DDN Data Intelligence Platform / Infinia: unified AI data platform, metadata indexing, multi-tenancy, multi-protocol support, AI inference, analytics, data preparation, model loading, security.
- DDN EXAScaler: high-performance parallel file system for AI/HPC, GPU efficiency, training, checkpointing, large-scale workloads.
- DDN + NVIDIA reference architecture themes: NVIDIA DGX/HGX/SuperPOD-style AI infrastructure, AI400X-class storage, EXAScaler, Infinia, NVIDIA AI Enterprise, high-speed networking/fabric.

## Guardrails
Do not claim DDN has a one-to-one fraud/risk/trading BOM unless source-backed content explicitly exists.
Do not output exact DDN SKUs as a recommendation.
Do not output node counts, GPU counts, pricing, or final sizing.
Do not say “DDN says this exact workload requires X.”

Use safer wording:

```text
Based on DDN public FSI and AI reference architecture material, this workload most closely aligns to...
The captured inputs point toward...
This resembles a DDN-informed pattern around...
This should be validated against DDN/reference architecture sizing guidance and customer constraints before a BOM.
```

## Implementation approach
Add a clean correlation data structure, preferably:

```text
ui/partner-hub/components/workload-mapper/workload-mapper-ddn-reference.ts
```

or similar.

Each workload should include:

```ts
workloadId
closestDdnPattern
mappingConfidence
sourceBasis
buildingBlocks
validationBeforeBom
summaryGuidance
```

Each building block should include:

```ts
name
whatItDoes
whyItFits
capturedInputSignals
validationQuestions
```

## Workload correlation map
Use the following mapping as the baseline.

### Fraud Detection & Investigation
Closest pattern:
Real-time AI data pipeline + inference + governed analytics.

Building blocks:
- Unified fraud data ingestion pipeline
- High-throughput / low-latency active data platform
- Metadata / indexing layer
- AI anomaly detection and scoring layer
- Real-time inference layer
- Investigator context / retrieval layer
- Governance, security, and compliance layer
- HA/DR and replication layer
- Performance validation / benchmark plan

Why:
Fraud workloads need to correlate transaction, identity, behavioral, historical, and case data quickly; support real-time or near-real-time scoring; provide investigator context; protect regulated data; and validate event-to-score/case-context timing.

### Risk Modeling / Stress Testing
Closest pattern:
HPC/analytics data pipeline + high-performance shared data.

Building blocks:
- Governed historical risk data platform
- Scenario data pipeline
- Parallel read / analytics execution layer
- Scenario output and result aggregation layer
- Governance, lineage, and reproducibility controls
- Performance validation for batch/intraday run windows

Why:
Risk/stress workflows depend on large historical datasets, scenario runs, reproducibility, result aggregation, and compressing risk windows.

### Trading / Quant Research
Closest pattern:
HPC/AI analytics + low-latency active data pattern.

Building blocks:
- Time-series / feature data active platform
- High-concurrency research access layer
- Metadata / indexing layer for feature and backtest artifacts
- Backtesting / analytics execution layer
- Low-latency data access validation
- Governance controls for proprietary strategy data

Why:
Quant/trading research needs fast access to market/feature data, high researcher/job concurrency, low-latency hot reads, and rapid backtesting iteration.

### Compliance / Document Intelligence
Closest pattern:
Governed RAG / document intelligence pattern.

Building blocks:
- Governed document ingest pipeline
- Chunking / embedding pipeline
- Metadata indexing layer
- Vector / retrieval layer
- Explainability and citation/evidence layer
- Security, residency, audit, and access-control layer

Why:
Compliance workloads need document ingestion, metadata/indexing, retrieval, explainability, auditability, and governed access to sensitive regulatory/policy/control evidence.

### RAG Knowledge Assistant
Closest pattern:
Enterprise RAG / inference / data intelligence pattern.

Building blocks:
- Enterprise knowledge ingestion pipeline
- Metadata and indexing layer
- Embedding / vector retrieval layer
- Model serving / inference layer
- RBAC/ABAC governed access layer
- Retrieval freshness validation

Why:
RAG assistants need many-doc ingestion, chunking/embedding, retrieval freshness, governed access, and low-latency inference.

### Model Training / Fine-tuning
Closest pattern:
DDN + NVIDIA AI reference architecture / DGX-HGX + EXAScaler training pattern.

Building blocks:
- Curated training data platform
- High-throughput training data layer
- GPU training / fine-tuning compute layer
- Checkpoint and artifact layer
- Model registry / evaluation workflow
- High-speed fabric / data movement validation
- Benchmark and sizing validation before BOM

Why:
Training/fine-tuning depends on model size, dataset size, batch size, training window, checkpoint cadence, GPU topology, and data path efficiency.

### Large-scale AI Training
Closest pattern:
NVIDIA HGX/DGX/SuperPOD + DDN AI400X/EXAScaler validated AI factory pattern.

Building blocks:
- Distributed corpus / preprocessing pipeline
- EXAScaler-style high-performance parallel training data layer
- GPU training cluster / AI factory compute layer
- Checkpointing and artifact management
- High-speed interconnect / fabric
- AI Enterprise/MLOps orchestration
- GPU utilization and throughput benchmark plan

Why:
Large-scale training needs sustained throughput, checkpoint efficiency, distributed preprocessing, accelerator utilization, and validated fabric/storage/compute balance.

### Inference Platform
Closest pattern:
Infinia + AI inference / data-intelligence pattern.

Building blocks:
- Model serving tier
- Cache / session state layer
- Metadata/indexing and retrieval support
- Low-latency active data layer
- Observability / guardrails
- Multi-tenant governance/security
- Latency/concurrency validation

Why:
Inference platforms depend on p95/p99 latency, request concurrency, model size, token/context volume, cache strategy, and HA/serving design.

### Scientific Simulation
Closest pattern:
HPC simulation / parallel file system pattern.

Building blocks:
- Scheduler-integrated HPC compute fabric
- EXAScaler-style parallel file/data platform
- Checkpoint / output handling layer
- High-speed interconnect
- Post-processing / visualization workflow
- Job profile and checkpoint validation

Why:
Simulation workloads need scheduler integration, parallel file access, checkpoint/output handling, fabric performance, and post-processing workflows.

### Genomics / Life Sciences
Closest pattern:
Metadata-heavy HPC/AI analytics pattern.

Building blocks:
- Sequencing/pipeline ingest layer
- Metadata/indexing layer for many small files
- High-throughput pipeline data layer
- Analytics / AI interpretation layer
- Governance, retention, and reproducibility layer
- Regulated data access controls

Why:
Genomics workflows often involve many small files, pipeline throughput, long retention, reproducibility, and regulated/consent-aware controls.

### Media / Rendering Pipeline
Closest pattern:
High-throughput shared data + inference/rendering pattern.

Building blocks:
- High-throughput shared asset platform
- Metadata/indexing layer for media assets
- GPU render/inference pool
- Workflow orchestration / queue layer
- Production burst ingest and throughput validation
- Project-level access controls and archive policy

Why:
Media/rendering workflows involve large mixed files, production ingest bursts, GPU render/inference jobs, deadline-driven scale, and asset metadata/indexing.

### Custom Workload
Closest pattern:
Derive from selected AI pattern and captured fields.

If RAG: use enterprise RAG/data intelligence pattern.
If Training/Fine-tuning: use AI training/reference architecture pattern.
If Inference: use inference/data intelligence pattern.
If Analytics/ML: use analytics/high-performance data platform pattern.
If HPC simulation: use parallel file/HPC pattern.

## BOM Summary output requirement
The BOM Summary should include:

1. Opening disclaimer:
```text
This is not a final BOM. It is a DDN-informed architecture mapping based on the current workload inputs.
```

2. Closest pattern:
```text
Based on the captured inputs, this workload most closely aligns to...
```

3. Building blocks:
For each relevant building block, explain:
- what it does
- why this workload needs it
- which captured input points to it

4. Validation before BOM:
Explain what still needs validation before a real BOM.

5. Closing note:
```text
This is an example starting point based on current DDN public reference architecture and platform themes. Actual sizing and final BOM require DDN/customer-specific validation, benchmark expectations, and constraints.
```

## UI/LLM behavior
Feed the DDN reference-pattern object into the BOM Summary prompt along with captured workload fields.

The local LLM should use:
- selected workload fields
- BOM input capture values
- DDN reference-pattern mapping
- building blocks
- validation-before-BOM list

The model should not invent extra building blocks if the mapping already provides them.

## Validation
Run:

```bash
npm run lint
npm run typecheck
npm test -- --runInBand
npm run build
rg -i "chip-?8|steam[- ]?trains" . || true
```

## Acceptance criteria
- BOM Summary maps each workload to a DDN-informed reference pattern.
- BOM Summary explains building blocks specifically.
- BOM Summary ties building blocks to captured inputs.
- BOM Summary includes validation-before-BOM guidance.
- BOM Summary includes clear disclaimer/closing note.
- No exact BOM, SKU, node count, GPU count, or pricing is generated.
- Works for all preloaded workloads and custom workload.
- No stale CHIP-8 / Steam Trains references.
